pkg/dataobj/internal/dataset/page_builder.go

```diff
@@ -57,8 +57,15 @@ type pageBuilder struct {
 // combination of opts.Value and opts.Encoding.
 func newPageBuilder(opts BuilderOptions) (*pageBuilder, error) {
 	var (
+		// We avoid giving an initial capacity to the buffers below to reduce
+		// the memory footprint for page builders (one per column), since it's
+		// atypical for every column to completely fill its pages.
+		//
+		// However, this will slightly increase the memory footprint for any
```

Comment:
If this becomes a problem we could write a custom resizable buffer with a maximum size, but I didn't feel like doing that right now 😛